mender/templates/api-gateway/hpa.yaml

```diff
@@ -0,0 +1,32 @@
+{{- if .Values.api_gateway.hpa.enabled }}
+apiVersion: autoscaling/v2
+kind: HorizontalPodAutoscaler
+metadata:
+  name: {{ include "mender.fullname" . }}-api-gateway
+spec:
+  scaleTargetRef:
+    apiVersion: apps/v1
+    kind: Deployment
+    name: {{ include "mender.fullname" . }}-api-gateway
+  minReplicas: {{ .Values.api_gateway.replicas }}
+  maxReplicas: {{ .Values.api_gateway.hpa.maxReplicas }}
+  metrics:
+{{- if .Values.api_gateway.hpa.cpu }}
+  - type: Resource
+    resource:
+      name: cpu
+      target: 
+{{ toYaml .Values.api_gateway.hpa.cpu.target | indent 8 }}
+{{- end }}
+{{- if .Values.api_gateway.hpa.memory }}
+  - type: Resource
+    resource:
+      name: memory
+      target: 
+{{ toYaml .Values.api_gateway.hpa.memory.target | indent 8 }}
+{{- end }}
+{{- if .Values.api_gateway.hpa.behavior }}
+  behavior:
+{{ toYaml .Values.api_gateway.hpa.behavior | indent 4 }}
+{{- end }}
+{{- end }}
```

Comment:
May I suggest to put this into a reusable templates to avoid future errors when updating the templates.
```yaml
# _helpers.yaml
---
{{-/*
Synopsis:
{{ include "mender.hpa" (dict 
    "hpa" .Values.<service>.hpa 
    "name" ( printf "%s-%s (include "mender.fullname" .) ) 
) }}
{{- if .hpa.enabled }}
apiVersion: autoscaling/v2
kind: HorizontalPodAutoscaler
metadata:
  name: {{ .name }}
spec:
  scaleTargetRef:
    apiVersion: apps/v1
    kind: Deployment
    name: {{ .name }}
  minReplicas: {{ .hpa.minReplicas }}
  maxReplicas: {{ .hpa.maxReplicas }}
  metrics:
  {{- if .hpa.metrics }}
    {{- toYaml .hpa.metrics | nindent 4 }}
  {{- end }}
  {{- if .hpa.behavior }}
  behavior:
    {{- toYaml .hpa.behavior | nindent 4 }}
  {{- end }}
{{- end }}
 */-}}
```

Reply:
Really nice stuff! Thanks for the hint